hooks/templatedatavalidator.js

```diff
@@ -25,13 +26,50 @@ module.exports = function (pageData, partials) {
  * Validates global config for the page template
  * 
  * @param {Object} globalConfig 
+ * @param {Object} JAMBO_INJECTED_DATA
  * @returns {boolean}
  */
-function isGlobalConfigValid(globalConfig) {
-  if (!globalConfig.experienceKey) {
+function isGlobalConfigValid(globalConfig, JAMBO_INJECTED_DATA) {
+  const { experienceKey } = globalConfig;
+  if (!experienceKey) {
     error('Missing Info: no experienceKey found.');
     return false;
   }
+  if (globalConfig.useJWT || globalConfig.apiKey) {
+    return true;
+  }
+
+  const injectedDataForExperience = JAMBO_INJECTED_DATA.answers.experiences[experienceKey];
+  if (!injectedDataForExperience) {
+    error(`No JAMBO_INJECTED_DATA found for experience key: "${experienceKey}"`);
+    error(`Found JAMBO_INJECTED_DATA: "${JSON.stringify(JAMBO_INJECTED_DATA, null, 2)}.`);
+    return false;
+  }
+
+  const productionApiKey = injectedDataForExperience.configByLabel.PRODUCTION.apiKey;
+  const deprecatedApiKey = injectedDataForExperience.apiKey;
+  if (!productionApiKey) {
+    if (!deprecatedApiKey) {
+      error(`No injected production api key found for experience key: "${experienceKey}"`);
+    } else {
+      warn('No injected production api key found, using the default apiKey instead.');
+    }
+  }
+
+  const stagingApiKey = injectedDataForExperience.configByLabel.STAGING.apiKey;
+  if (!stagingApiKey) {
```

Comment:
It'd be nice if we had access to `isStaging` here, so we knew which of these blocks to care about. I'm wondering if we need something like that here. What about the case where isStaging is true and we can't find an `apiKey` for the `PRODUCTION` env. Does that really matter? We can still build the `STAGING` pages.

Reply:
isStaging is determined by the browser's URL so unfortunately we can't use it at build time

Reply:
I'm asking Max if it's possible to ever have one version without the other. I wouldn't want to have an overly harsh validator.

Reply:
Okay, sounds like you can never have one without the other! So we're good here.